lib/app/app.rb

```diff
@@ -83,53 +84,90 @@ class ConnectorsWebApp < Sinatra::Base
   end
 
   post '/status' do
-    source_status = @connector.source_status(body_params)
+    connector = settings.connector_class.new
+
+    source_status = connector.source_status(body_params)
     json(
-      :extractor => { :name => @connector.name },
+      :extractor => { :name => connector.name },
       :contentProvider => source_status
     )
   end
 
   post '/documents' do
-    results, cursors, completed = @connector.document_batch(body_params)
```

Comment:
Yah, related to a similar comment in the other PR, I think we need to respect `cursors` at the beginning and end of the sync.

Reply:
That's a good point.

Big question for me - how much does it matter now since we're switching to a different approach to take.

cc @tarekziade @serenachou

Reply:
I don't think we can consider the async efforts here complete without preserving `cursors` behavior. We don't need to handle mid-sync cursors like we had introduced for SharePoint Online, but if the final cursors value is not reported back to Enterprise Search after a SharePoint Online incremental sync, I think every incremental sync would start to take about a full sync's amount of time.

Reply:
yeah I also think we should keep the existing behavior at the API level for now, even if the transient cursor is not really used